docs/porting/target/cmake.md

```diff
@@ -0,0 +1,124 @@
+
+- `MBED_CONFIG_DEFINITIONS`: a list of macro definitions added to a given Mbed target coming from various Mbed library configuration files.
+- `MBED_TARGET_SUPPORTED_C_LIBS`: a list of C library types supported by a given Mbed target coming from the Mbed target default configuration file `targets.json`.
+- `MBED_TARGET_SUPPORTED_APPLICATION_PROFILES` a list of application profiles supported by a given Mbed target coming from the Mbed target default configuration file `targets.json`.
+
+## Mbed targets CMake input file structure
+
+### Vendor selection
+
+Vendor selection is handled in [`targets/CMakeLists.txt`](LINK) using a conditional statement to verify that one of the labels used by a given Mbed target match a given partner directory. The matching directory is added to directories to build.
+
+```
+if("Cypress" IN_LIST MBED_TARGET_LABELS)
+    add_subdirectory(TARGET_Cypress)   
+endif()
+```
+Add an `elseif` conditional statement to add a vendor directory.
+### MCU family targets
+
+`targets/TARGET_<VENDOR_NAME>` usually contains different MCU or Mbed target families. The structure of the vendor directory is up to the vendor but it is preferable that it has logical separation separations.
+For example, the content of `targets/TARGET_Cypress/TARGET_PSOC6` can be listed as follows:
+
+```
+if("SCL" IN_LIST MBED_TARGET_LABELS)
+    add_subdirectory(COMPONENT_SCL EXCLUDE_FROM_ALL)
+endif()
+
+if("WHD" IN_LIST MBED_TARGET_LABELS)
+    add_subdirectory(COMPONENT_WHD EXCLUDE_FROM_ALL)
+    add_subdirectory(common/COMPONENT_WHD EXCLUDE_FROM_ALL)
+endif()
+
+if("CY8CKIT064B0S2_4343W" IN_LIST MBED_TARGET_LABELS)
+    add_subdirectory(TARGET_CY8CKIT064B0S2_4343W)
+elseif("CY8CKIT_062S2_43012" IN_LIST MBED_TARGET_LABELS)
+    add_subdirectory(TARGET_CY8CKIT_062S2_43012)
+...
+endif()
+
+# Add the include directories accessible from this directory that are not specific to an MCU family or Mbed target
+target_include_directories(mbed-core
+    INTERFACE
+        ...
+)
+
+# Add the source files accessible from this directory that are not specific to an MCU family or Mbed target
+target_sources(mbed-core
+    INTERFACE
+        ...
+)
+```
+
+## Add target's CMakeLists.txt
+
+Add `CMakeLists.txt` files to the top level directory of a given vendor directory. List all files found in the directory in this CMake input source file, adding additional CMake input source file if it is MCU or Mbed target specific and has a great number of files which will make the top level `CMakeLists.txt` too complex. Think when you decide to create functions in a computer software code to remove complexity.
+
+See [`targets/TARGET_Cypress`](LINK) for examples.
+
+### Add your sources and includes
+
+A whole directory can be added to the build using `add_subdirectory` if it is specific to an MCU family (or Mbed target) and has a complex structure containing more directories and files. Otherwise simply list the files contained in the directory.
```

Comment:
```suggestion
A whole directory can be added to the build using `add_subdirectory()` if it is specific to an MCU family (or Mbed target) and has a complex structure containing more directories and files. Otherwise simply list the files contained in the directory.
```